src/Network/Network/Models/PSSubnet.cs

```diff
@@ -41,6 +41,10 @@ public class PSSubnet : PSChildResource
         [Ps1Xml(Label = "RouteTable Name", Target = ViewControl.Table, ScriptBlock = "$_.RouteTable.Name")]
         public PSRouteTable RouteTable { get; set; }
 
+        [JsonProperty(Order = 1)]
+        [Ps1Xml(Label = "NatGateway Name", Target = ViewControl.Table, ScriptBlock = "$_.NatGateway.Name")]
+        public PSNatGateway NatGateway { get; set; }
```

Comment:
Basing on the [specs](https://github.com/Azure/azure-rest-api-specs/blob/network-march-release/specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetwork.json#L1053) this need to be of type `PSResourceId`. Otherwise there would be type mapping failure